flake.nix

```diff
@@ -0,0 +1,45 @@
+{
+  description = "A very basic flake for Rust development";
+
+  inputs.std.url = "github:divnix/std/v0.24.0-1";
+  inputs.std.inputs.nixpkgs.follows = "nixpkgs";
+
+  inputs.fenix.url = "github:nix-community/fenix";
+  inputs.crane.url = "github:ipetkov/crane";
+  inputs.crane.inputs.nixpkgs.follows = "nixpkgs";
+  inputs.crane.inputs.flake-compat.follows = "";
+  inputs.crane.inputs.rust-overlay.follows = "";
+
+  inputs.std.inputs.devshell.url = "github:numtide/devshell";
+
+  inputs.nixpkgs.follows = "fenix/nixpkgs";
```

Comment:
Mostly for my knowledge, can you fill me in on why these are all necessary?

Why fenix + crane? (I usually have used fenix to get a nightly toolchain, is that somehow needed here)?

What does `std` provide?

Reply:
Sorry I didn't even think much about the Nix, I just used the Rust template that I made for Standard, that I am used to, but I can remove the Nix honestly. That was mostly for my own use, but it does setup a nice devshell with rust-analyzer ready to go for those who use terminal based editors.

Reply:
doesn't bother me, especially if you're maintaining it :)